### PR TITLE
Create node.js.yml

### PR DESCRIPTION
**Pull request by P0NDER0SA**
Adding the start for github actions.  it has the basic node setup/validation and some testing.  i've got the cypress section commented out until we actually include it.

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Node.js CI
+
+on:
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
```
[the remark this answers is not in the document]
Comment (P0NDER0SA) [reply .github/workflows/node.js.yml:11]: ya, sure!

File: .github/workflows/runtests.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build_and_unit_test:
+  #Unit Tests
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [10.x, 12.x, 14.x, 15.x]
```
Comment (var-kyle) [inline .github/workflows/runtests.yml:16]: I would remove version 10 and 15, since 10 is EOL at the end of April, and 15 isn't an LTS release.

**Review by var-kyle — CHANGES_REQUESTED** [review]

File: .github/workflows/runtests.yml (hunk 2 above)
Comment (P0NDER0SA) [reply .github/workflows/runtests.yml:16]: :)  alright

**Review by var-kyle — APPROVED** [review]
Looks good!

**Pull request merged**